### DISCUSSION TITLE
Custom OAuthClientId and OAuthAppName  for each definition

### DISCUSSION THREAD

**Discussion by Egreg0r** [Burgyn/MMLib.SwaggerForOcelot#244] [Q&A] [answered]
Hi.
I have some API microservice with MS identity server autorization. Every microservice use PKCE. 
In the swagger API for fast authorization,  set options SwaggerUI. For example:
```
app.UseSwaggerUI(options =>
            {
                options.SwaggerEndpoint("v1/swagger.json", "OrderBook Api - Swagger");
                options.OAuthClientId("orderbook");
                options.OAuthAppName("Orderbook");
                options.OAuthUsePkce();
            });
```
And in Ui returns:
![2022-11-17_102339](https://user-images.githubusercontent.com/28818687/202412772-2b1ff280-4b9c-41f4-aea9-8f6f8bf9cb54.png)

In Ocelot swagger, i have one definition for each api micrioservice.
But if write:
```
           app.UseSwaggerForOcelotUI(options =>
            {
                options.PathToSwaggerGenerator = "/swagger/docs";
                options.OAuthUsePkce();
                options.OAuthClientId("orderbook");
                options.OAuthAppName("Orderbook");
            })
                .UseOcelot()
                .Wait();
```
Will return ClientId = "orderbook" for all definition.
But need to fill in the appropriate client for every definition. 
Is there a way to specify clientId for each definition? 
For example in `ocelot.json` in `"SwaggerEndPoints"` set `"OAuthClientId":"orderbook"`

**Comment by Burgyn** [comment, marked as answer]
Hi @Egreg0r,

Unfortunately, there is no direct way to do this.
Please see this discussion #241 

Thanks.
